framework/include/base/FEProblem.h

```diff
@@ -481,12 +481,12 @@ class FEProblem :
    * @return Const reference to the user object
    */
   template <class T>
-  const T & getUserObject(const std::string & name)
+  const T & getUserObject(const std::string & name, unsigned int tid = 0)
```

Comment:
Yeah, I guess I don't really see the reason for this, as we are staying in the **same** thread in this design. I'm also not asking for specific data, I'm merely relying on accessing the initialized element in that thread. 

Again, no cross-thread access, no internal state of the userobject that is prepared by MOOSE.